### PR TITLE
Honda Odyssey 2018 EX-L with Honda Sensing

### PR DESCRIPTION
**Pull request by energee**
_Just needs to be tested_

### CODE REVIEW

**Comment by geohot** [comment]
Wow, that was fast. Will merge after we get some testing.

File: selfdrive/car/honda/interface.py [hunk 1]
```diff
@@ -200,6 +200,19 @@ def get_params(candidate, fingerprint):
...
 
+      ret.longitudinalKpBP = [0., 5., 35.]
+      ret.longitudinalKpV = [1.2, 0.8, 0.5]
+      ret.longitudinalKiBP = [0., 35.]
+      ret.longitudinalKiV = [0.18, 0.12]
+    elif candidate == "HONDA ODYSSEY 2018 EX-L":
+      stop_and_go = False
+      ret.m = 4354./2.205 + std_cargo
+      ret.l = 3.00
+      # Wtf are these?
```
Comment (rbiasini) [inline selfdrive/car/honda/interface.py]: Allright, I'll change those var names into more explicit ones. 
Definitions are in cereal/car.capnp

File: selfdrive/car/honda/interface.py [hunk 2]
```diff
@@ -200,6 +200,19 @@ def get_params(candidate, fingerprint):
       ret.sR = 15.3
       ret.steerKp, ret.steerKi = 0.8, 0.24
 
+      ret.longitudinalKpBP = [0., 5., 35.]
+      ret.longitudinalKpV = [1.2, 0.8, 0.5]
```
Comment (rbiasini) [inline selfdrive/car/honda/interface.py]: Is the brake system like the one on the civic or like the one on the acura ILX you think? Can you hear constant pump noise when brakes are applied at low speed? If so, it's probably like the acura and this tuning is fine.

File: selfdrive/car/honda/interface.py (hunk 1 above)
Comment (geohot) [reply selfdrive/car/honda/interface.py]: Long names!

Comment (rbiasini) [reply selfdrive/car/honda/interface.py]: fixed. @energee, will merge in openpilot devel soon.

**Comment by Lucianolps1** [comment]
will the system work with a new odyssey 2018 or 2017?

**Comment by energee** [comment]
This was written for the 2018.

**Comment by Lucianolps1** [comment]
We currently own a Toyota sienna 2015, i am looking to trade for a Odyssey or pilot, both are equiped with the Honda Sensing.  Will the system work also for the Pilot?
Thank you

**Comment by alan707** [comment]
@energee what is the difference between 2018 and 2017 Honda Odyssey?

**Comment by energee** [comment]
@alan707 I'm not sure if there is one, 2017 has ACC/LKAS?

**Comment by alan707** [comment]
@energee amazing work on this, BTW.  You are right. Looks like the 2017 Honda Odyssey doesn't have Honda Sensing.

**Comment by energee** [comment]
No, only Hondas with stop and go are Civic, 2017+ CR-V, and 2018 Accord

**Comment by alan707** [comment]
@energee thanks for the reply.  I assume you would need an interceptor pedal to make that happen, right?  I couldn't find any info on the comma forums or the Github issues.  Any idea on how to convert it into stop and go?

**Comment by alan707** [comment]
@energee also do you know what is the main difference between the Civic, CR-V and 2018 Accord?  Is that functionality within the Engine Control Module or Body Control Module firmware, you think?

**Comment by energee** [comment]
I think its the power-train control module/engine control module where this limitation exists. I don't understand why this OEM limitation exists on several of the Hondas, must be due to poor object detection at low speeds, but the civic shares very similar radar hardware.

You are correct about interceptor pedal, they are doing this on comma's development ILX. It might be possible to enable this control via firmware mod, though.

**Comment by rbiasini** [comment]
@energee this PR looks good. Let me know how tests go and we'll merge!

**Comment by alan707** [comment]
@energee do you have any additional info about the interceptor pedals?

**Comment by energee** [comment]
I don't know anything about hardware mods, just software

**Comment by mwkrieger** [comment]
This looks awesome! How do I get this on my new EON?

**Comment by energee** [comment]
@mwkrieger Testing right now, we are close.

**Comment by energee** [comment]
Ok this is working now, j4z (slack) has the car. @rbiasini, He said braking is rough and makes a noise, wasn't there a setting to address this?

**Comment by rbiasini** [comment]
@energee brake pump noise is probably the same as the Acura. Isn't the CRV pump noisy too?
A slightly better solution (like the stock system) would be to only run the pump when the brake pressure is required to increase.
Braking shouldn't be rough tough, but it's possible that the Pilot has a different scale from the Acura/CRV. It might require tuning

**Comment by energee** [comment]
I have the newer crv, not 2016, seems like my hardware is closest to the civic. Ill let j4z chime in here. Any suggestions on which direction to scale these? Btw this one is odyssey not pilot.

**Comment by joeljacobs** [comment]
@energee @rbiasini  Hi, I don't know what to experiment with to work on the brakes. Right now they are quite jerky - like they are being pressed harder than they should, but for short times. What parameter in which file should I mess with? I looked around in the carcontroller.py, but wasn't sure which was the right parameter.

**Comment by rbiasini** [comment]
@joeljacobs @energee , I would try using the civic tuning then. Civic has indeed a steeper relation between brake CAN command and vehicle decel. In particular,  make sure you use `the compute_gb_honda` function as opposed to the one for the Acura.

ultimately, you should find out the specific relation between CAN brake command and vehicle decel, which is usually pretty much linear with a small offset.

**Comment by energee** [comment]
@rbiasini This is ready to be committed, https://github.com/commaai/opendbc/pull/61 is dependency.

**Comment by rbiasini** [comment]
@energee, looks good! can we checkout some of the items in the list I've sent you on slack?
And yes, will merge the dbc changes, just waiting on a minor style change

**Comment by mwkrieger** [comment]
Here is the full fingerprint that worked with my 2018 Odyssey Elite. Would be great if it was integrated as well. 

57L: 3, 148L: 8, 228L: 5, 229L: 4, 304L: 8, 342L: 6, 344L: 8, 380L: 8, 399L: 7, 411L: 5, 419L: 8, 420L: 8, 427L: 3, 432L: 7, 440L: 8, 450L: 8, 463L: 8, 464L: 8, 476L: 4, 490L: 8, 506L: 8, 507L: 1, 542L: 7, 545L: 6, 597L: 8, 662L: 4, 773L: 7, 777L: 8, 780L: 8, 795L: 8, 800L: 8, 804L: 8, 806L: 8, 808L: 8, 817L: 4, 819L: 7, 821L: 5, 825L: 4, 829L: 5, 837L: 5, 856L: 7, 862L: 8, 871L: 8, 881L: 8, 882L: 4, 884L: 8, 891L: 8, 892L: 8, 905L: 8, 923L: 2, 927L: 8, 929L: 8, 963L: 8, 965L: 8, 966L: 8, 967L: 8, 983L: 8, 985L: 3, 1029L: 8, 1036L: 8, 1052L: 8, 1064L: 7, 1088L: 8, 1089L: 8, 1092L: 1, 1108L: 8, 1110L: 8, 1125L: 8, 1296L: 8, 1302L: 8, 1600L: 5, 1601L: 8, 1612L: 5, 1613L: 5, 1614L: 5, 1616L: 5, 1619L: 5, 1623L: 5, 1668L: 5, 416944369L: 8, 417001744L: 8,

File: selfdrive/car/honda/carcontroller.py [hunk 3]
```diff
@@ -121,8 +121,8 @@ def update(self, sendcan, enabled, CS, frame, actuators, \
     tt = sec_since_boot()
     GAS_MAX = 1004
     BRAKE_MAX = 1024/4
-    if CS.civic:
-      is_fw_modified = os.getenv("DONGLE_ID") in ['b0f5a01cf604185c']
+    if CS.civic or CS.odyssey:
+      is_fw_modified = os.getenv("DONGLE_ID") in ['b0f5a01cf604185cxxx']
```
Comment (rbiasini) [inline selfdrive/car/honda/carcontroller.py]: can you remove xxx? @energee

**Comment by geohot** [comment]
Congrats!

**Comment by energee** [comment]
Nice! Thank you

**Pull request merged**